conans/test/functional/toolchains/meson/test_cross_x86.py

```diff
@@ -29,6 +29,9 @@ def config_options(self):
                 if self.settings.os == "Windows":
                     del self.options.fPIC
 
+            def layout(self):
```

Comment:
Meson needs a clean folder, it was hardcoding "build" in ``Meson`` helper directly. Now it requires defining ``layout()`` better than that.

Reply:
see #9713. it's still unclear how to use layout to get reproducible build directories.

Reply:
It is not unclear, it is very well defined that the layout should be defined by ``layout()``, and not rely on ``Meson()`` locally defining a ``build`` folder. Whether we need more of the ``layout()`` and we could use a ``meson_layout()``, is totally possible, but we are certain that we don't want this in the Meson helper.